pom.xml

```diff
@@ -16,51 +18,42 @@
 
     <artifactId>Office-365-Connector</artifactId>
     <name>Office 365 Connector</name>
-    <version>4.17.1-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <description>This plugin sends Job notifications to JenkinsCI Office 365 Connector.</description>
     <url>https://github.com/jenkinsci/office-365-connector-plugin</url>
 
     <properties>
-        <powermock.version>1.6.6</powermock.version>
-        <jenkins.version>2.277.2</jenkins.version>
-        <java.level>8</java.level>
+        <revision>4.17.1</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <mockito.version>3.12.4</mockito.version>
+        <powermock.version>2.0.9</powermock.version>
+        <jenkins.version>2.319.3</jenkins.version>
+        <gitHubRepo>jenkinsci/office-365-connector-plugin</gitHubRepo>
     </properties>
 
-    <dependencies>
-        <!-- fixes error:
-             Failed while enforcing RequireUpperBoundDeps -->
-        <dependency>
-            <groupId>commons-beanutils</groupId>
-            <artifactId>commons-beanutils</artifactId>
-            <version>1.9.4</version>
-            <scope>provided</scope>
-        </dependency>
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.319.x</artifactId>
+                <version>1654.vcb_69d035fa_20</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
 
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.13.1</version>
-            <scope>test</scope>
-        </dependency>
+    <dependencies>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>1.10.19</version>
             <scope>test</scope>
-            <!-- powermock wants objenesis 2.4 and mockito wants objenesis 2.1 -->
-            <!-- exclude from mockito so that powermock version is used -->
-            <exclusions>
-                <exclusion>
-                    <groupId>org.objenesis</groupId>
-                    <artifactId>objenesis</artifactId>
-                </exclusion>
-            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
-            <version>2.8.0</version>
+            <version>2.9.1</version>
```

Comment:
This PR is a big change and I'm not happy to review and accept it as once. 
Suggest to start from updating this library and powermock. Then we can work on other parts

Reply:
#286

Reply:
1 & 2 have been merged - thanks for PR !